corehq/apps/ota/views.py

```diff
@@ -69,22 +70,22 @@ def search(request, domain):
 
 
 @csrf_exempt
+@require_POST
 @json_error
 @login_or_digest_or_basic_or_apikey()
 def claim(request, domain):
     couch_user = CouchUser.from_django_user(request.user)
     case_id = request.POST['case_id']
-    if request.method == 'POST':
-        if request.session.get('last_claimed_case_id') == case_id:
-            return HttpResponse('You have already claimed that {}'.format(request.POST.get('case_type', 'case')),
-                                status=400)
-        try:
-            claim_case(domain, couch_user.user_id, case_id,
-                       host_type=request.POST.get('case_type'), host_name=request.POST.get('case_name'))
-        except CaseNotFound:
-            return HttpResponse('The case "{}" you are trying to claim was not found'.format(case_id),
-                                status=404)
-        request.session['last_claimed_case_id'] = case_id
+    if request.session.get('last_claimed_case_id') == case_id:
```

Comment:
I know you didn't just add this, but what happens if you've already claimed this case in the past (i.e. not just in the last session)? This could be another failure mode of the `claim_case` function.